### PR TITLE
Fix checkpoint loading when model_name contains models or additional components

### PR DESCRIPTION
**Pull request by gablanouette**
Updates to the model_from_checkpoint function to ensure handling of efficientdet based on various ways to name models

### CODE REVIEW

**Comment by gablanouette** [comment]
updates to model_from_checkpoint function to be able to handle efficientdet models

**Comment by FraPochetti** [comment]
Hi @gablanouette I already see that the formatting check is failing.
You need to run `black`.
You can do that automatically using `pre-commit hooks`.
Check them [here](https://airctic.com/0.11.0/contributing/).

File: icevision/models/checkpoint.py [hunk 1]
```diff
@@ -208,8 +205,11 @@ def model_from_checkpoint(
 
     model_type = None
     if model_name:
-        lib, mod = model_name.split(".")
```
Comment (FraPochetti) [inline icevision/models/checkpoint.py:211]: I see `lib` was instantiated before but it's not instantiated anymore in your PR.
Was it used somewhere else?
Any reason behind its removal?

Comment (gablanouette) [reply icevision/models/checkpoint.py:211]: It's no longer a binary splits so i do not believe they are needed any longer

**Comment by FraPochetti** [comment]
Pending https://discord.com/channels/735877944085446747/735877944517591151/922895633982062682

**Comment by gablanouette** [comment]
The code has been updated to add comments explaining why this is necessary to the users. For each case, if the length of the components in model_name is greater than 3, (e.g. models.mmdet.retinanet), we only want to use the library and model which are the second to last and last components. Otherwise, if model_name follows the default convention, the library and model are the first and second components, respectively (e.g. mmdet.retinanet)

**Comment by ai-fast-track** [comment]
Thanks a lot for your first contribution, @gablanouette! 🎉🥳

**Pull request merged**